test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -145,3 +171,73 @@ unsigned YPM::EnumTypeConverter<T>::enumMin()
 	else
 		static_assert(AlwaysFalse<T>::value, "Yul proto mutator: non-exhaustive visitor.");
 }
+
+template <typename T>
+void YPM::addControlFlow(T* _msg)
+{
+	enum class ControlFlowStmt: unsigned
+	{
+		For = 0,
+		BoundedFor,
+		If,
+		Switch,
+		FunctionCall,
+		Break,
+		Continue,
+		Leave,
+		Termination
+	};
+	uniform_int_distribution<unsigned> d(
+		static_cast<unsigned>(ControlFlowStmt::For),
+		static_cast<unsigned>(ControlFlowStmt::Termination)
+	);
+	auto random = static_cast<ControlFlowStmt>(d(s_rand.m_random));
+	Statement* s = basicBlock(_msg)->add_statements();
+	switch (random)
+	{
+	case ControlFlowStmt::For:
+		s->set_allocated_forstmt(new ForStmt());
+		break;
+	case ControlFlowStmt::BoundedFor:
+		s->set_allocated_boundedforstmt(new BoundedForStmt());
+		break;
+	case ControlFlowStmt::If:
+		s->set_allocated_ifstmt(new IfStmt());
+		break;
+	case ControlFlowStmt::Switch:
+		s->set_allocated_switchstmt(new SwitchStmt());
+		break;
+	case ControlFlowStmt::FunctionCall:
+		s->set_allocated_functioncall(new FunctionCall());
+		break;
+	case ControlFlowStmt::Break:
+		s->set_allocated_breakstmt(new BreakStmt());
+		break;
+	case ControlFlowStmt::Continue:
+		s->set_allocated_contstmt(new ContinueStmt());
+		break;
+	case ControlFlowStmt::Leave:
+		s->set_allocated_leave(new LeaveStmt());
+		break;
+	case ControlFlowStmt::Termination:
+		s->set_allocated_terminatestmt(new TerminatingStmt());
+		break;
+	}
+}
+
+template <typename T>
+Block* YPM::basicBlock(T* _msg)
+{
+	if constexpr (std::is_same_v<std::decay_t<T>, ForStmt>)
+		return _msg->mutable_for_body();
+	else if constexpr (std::is_same_v<std::decay_t<T>, BoundedForStmt>)
+		return _msg->mutable_for_body();
+	else if constexpr (std::is_same_v<std::decay_t<T>, SwitchStmt>)
+		return _msg->mutable_default_block();
```

Comment:
Just wondering if we ever mutate the case statement in switch?

Reply:
Ah, good catch. case stmt is missing here. But in general the fuzzer mutates everything in the spec. This PR is only adding a custom mutation to each "basic block" containing statement.

Reply:
Added case stmt, for init and post blocks, and code block i.e., `object { code { ... } }`